resteasy/src/main/java/com/cerner/beadledom/resteasy/ResteasyContextListener.java

```diff
@@ -41,7 +41,7 @@ public void contextInitialized(ServletContextEvent event) {
 
     final List<? extends Module> appModules = getModules(context);
 
-    List<Module> modules = Lists.newArrayList();
+    final List<Module> modules = new ArrayList<>();
```

Comment:
We shouldn't be making local variables final.

Reply:
Why not? It allows compiler to make some optimizations and makes a great case for the kind of defensive coding we should all indulge in. Frankly i find the lack of final keyword in the class signature, method signatures and transient variables not only a stark departure from my previous team, but also a little too tolerant

Reply:
Marking local variables as `final` doesn't provide much value and mostly just adds cruft to the method. If a method is sufficiently long that there is concern about a local variable being re-assigned, then the method should be split up; adding `final` is really just a band-aid for a bigger problem. The same is true of method parameters. Do you have an example of what compiler optimizations could be done for final local variables?

There sometimes is value in adding the `final` keyword to classes, but usually this would be for actual data classes. We usually solve this by using immutable data classes via autovalue. There may be some cases where marking methods as `final` is valuable but I haven't seen many.

There is definitely value in marking member variables as `final` since it indicates that no operations on that class will be re-assigning the member variables. This also indicates there is no need to worry about threading concerns related to those variables.

Following of strict process is not the same thing as creating high quality code. Adding `final` everywhere could be a process, but in reality it doesn't provide value in terms of code readability, maintainability, or defensiveness, for things like local variables and method parameters.

One thing I've found helpful in measuring my own code quality is to be able to have examples of high quality code to compare against. In Java I find the code of many of the Square open source libraries to be very high standards of good quality (e.g. https://github.com/square/retrofit and https://github.com/square/okhttp). No code-base is going to be perfect, but I would rather shoot for high quality code on principles and reason rather than by following a strict process.

Reply:
But are all those things mutually exclusive? Adding `final` wherever we can should not make the code any less readable or maintainable, nor does it violate any principles. If it does, then there are problems with the design. If we design something that is clearly NOT intended for extensibility or mutability, why would/should we even provide that flexibility for consumers (even if that consumers of something are just us) to potentially abuse our API? Isn't the functioning of our software under unforeseen circumstances like this the definition of defensive coding? I feel like having that mindset for local variables and method parameters is where it starts. And i don't understand how locking down our code in whatever way we can contributes in chipping away at code quality.

The above paragraph is just me providing a passionate defense for a mindset that has served me well. But at the end of the day, I've seen teams being stringent to a fault and teams being indifferent in this area and the scales aren't tipped in any particular direction in terms of the results. I'll remove the keyword.

Reply:
>  why would/should we even provide that flexibility for consumers (even if that consumers of something are just us) to potentially abuse our API?

How does having a final variable inside a method protect that method?

Reply:
That statement wasn't for the final variable, just extending the argument to classes that do not have a final keyword

Reply:
My previous statements mention that there is sometimes value in making a class final, but I'm not going to generalize it to say there is always value. That is one I would want to look at on a case-by-case basis.

Reply:
I'm with @johnlcox I think `final` has its uses and places that it will provide value. I'm just not a fan in blanket applying rules (some call it process) without any critical thinking about what makes sense for the current context.

Reply:
Making a class or a method parameter final when it is created specifically to be extended or re-assigned..THAT would be applying rules without any critical thinking, and thats not what i am advocating. I am saying that applying the rules when you're really indifferent about the extensibility of a class or mutability of a variable is okay and following this rule doesn't mean that it is senseless for a current context.

Again that being said, i am happy to remove the keyword for the code in this PR

Reply:
Also, why are local variables from 34-41 set to final? Shouldn't we remove the final keyword everywhere if we are this particular about enforcing it?

Reply:
I'm not sure why the variables on lines 34-41 are set to final. It would be good to fix those too.

Reply:
Looks like it's because checkstyle complains when the variable is defined far enough from its usage.  `Consider making that variable final if you still need to store its value in advance` is the fix that checkstyle is advocating.

Even if we re-order the definition of these variables in way that doesn't sacrifice readability, checkstyle still finds it problematic. If the fix for this is to retain the final keywords, in my opinion it creates more confusion because we'll be stuck in this weird place where some of them are final and some of them aren't and readers will not know why. We could change checkstyle settings but i think we all agree that that wouldn't really be a solution.

Reply:
It might be possible to rearrange and possibly split up the method, but if so it might be better to do as a separate PR.

Reply:
Like i said, re-arranging isn't possible without making it less readable. I tried re-arranging it for this particular use case.

Sorry for continuing to stretch this conversation further, but I am not convinced at all that we have to split up the method to get around something that (in my opinion) isn't a problem at all. We're trying to twist ourselves into knots unnecessarily when this method is absolutely fine the way it is.

Anyway, i've removed the final keyword: https://github.com/cerner/beadledom/commit/75d364f068be5b48f12a864aa2a8bd938fb929b8